src/book/02-system/01-foundations/03-continuous-integration.mdx

```diff
@@ -39,155 +49,31 @@ This CI process works in combination with the [Git Feature Branch workflow](http
-
-It is also possible that the [signature](https://wiki.archlinux.org/index.php/Pacman/Package_signing) on a package that is located on a public mirror changes. This change of signature can cause issues for us because we download date-locked packages and still expect an older signature. Caching such packages locally may alleviate these issues.
-
-To cache packages locally, a [pacman](https://wiki.archlinux.org/index.php/pacman) package cache is setup on each Buildkite agent. This cache uses the [nroi/flexo](https://hub.docker.com/r/nroi/flexo) Docker image, and exposes itself on:
-
-```
-http://localhost:7878/$repo/os/$arch
-```
-
-In order for [pacman](https://wiki.archlinux.org/index.php/pacman) to use this package cache, the cache must be added to `/etc/pacman.d/mirrorlist`. Because our build happens inside a [buildx](https://docs.docker.com/buildx/working-with-buildx/) container, we cannot use `localhost` as the address for the cache. Instead, we alias the IP address of the host machine as `host.docker.internal`. Therefore, the following is added to the mirrorlist [here in the Dockerfile](https://github.com/NUbots/NUbots/blob/5149f1c0cd1e4ff902d749269d1aa90f52adc075/docker/Dockerfile#L14):
-
-```
-Server = http://host.docker.internal:7878/$repo/os/$arch
-```
-
-## Setting Up a New Agent (Ubuntu)
-
-The following instructions show how to setup and configure a new Buildkite agent with Docker and Docker Compose. These steps have been collected into a [single script you can download here](assets/ci_setup.sh). Before running the script, ensure you have set the Buildkite agent token and DockerHub password correctly in the env variables at the top of the script. Also, make sure to run the script with `sudo`.
-
-### Install Docker
-
-Steps from <https://docs.docker.com/install/linux/docker-ce/ubuntu/>.
-
-```bash
-# Remove old Docker
-sudo apt-get remove docker docker-engine docker.io containerd runc
-
-# Install packages to allow apt to use a repository over HTTPS
-sudo apt-get update
-sudo apt-get install apt-transport-https ca-certificates curl gnupg-agent software-properties-common
-
-# Add Docker's official GPG key
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
-
-# Add Docker's stable repository
-sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
-​
-# Install latest Docker CE
-sudo apt-get update
-sudo apt-get install docker-ce docker-ce-cli containerd.io
-```
-
-### Install Docker Compose
+GitHub Actions are configured in [workflow files](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions), written in YAML. To add, remove, or change build jobs, we edit the relevant [workflow file in the NUbots repo](https://github.com/NUbots/NUbots/tree/7cc0677eda6629c2b445a35473748c163685aaad/.github/workflows). The [GitHub Actions documentation](https://docs.github.com/en/actions) should be your first stop when you want to make changes.
```

Comment:
This link also need to reference the main branch

Reply:
permalinks are the good stuff

Reply:
You want links to the commits rather than to main itself?

Reply:
+1 for permalinks but I'll grab a more recent commit at least. there's pros/cons to both but an out of date permalink is low risk low impact compare to a broken link imo.

Reply:
> You want links to the commits rather than to main itself?

If the path changes on main and you linked main, you have to start archaeology immediately to get any idea of the file contents. If you have a permalink, you can check main with a click, but you're guaranteed to get something useful even if it changed on main.

Reply:
Sure, if you say so

Reply:
We take that as a W